novawallet/Common/Services/RemoteSubscription/Substrate/Multisig/MultisigPendingOperationsSyncService.swift

```diff
@@ -0,0 +1,189 @@
+
+        let chainsFetchOperation = chainRepository.fetchAllOperation(with: .init())
+
+        execute(
+            operation: chainsFetchOperation,
+            inOperationQueue: operationQueue,
+            runningCallbackIn: workingQueue
+        ) { [weak self] result in
+            guard let self else { return }
+
+            switch result {
+            case let .success(chains):
+                let relevantChains = matchRelevantChains(from: chains)
+
+                mutex.lock()
+                pendingOperationsChainSyncServices = relevantChains.reduce(into: [:]) { acc, chain in
+                    let service = self.chainSyncServiceFactory.createMultisigChainSyncService(
+                        for: chain,
+                        selectedMetaAccount: selectedMetaAccount,
+                        knownCallData: self.knownCallData,
+                        operationQueue: self.operationQueue
+                    )
+                    acc[chain.chainId] = service
+                }
+                pendingOperationsChainSyncServices.forEach { $0.value.setup() }
+                mutex.unlock()
+            case let .failure(error):
+                logger?.error("Failed to fetch chains: \(error)")
+            }
+        }
+    }
+
+    func matchRelevantChains(from chains: [ChainModel]) -> [ChainModel] {
+        chains.filter { $0.hasMultisig }
+    }
+}
+
+// MARK: - MultisigPendingOperationsSyncServiceProtocol
+
+extension MultisigPendingOperationsSyncService: MultisigPendingOperationsSyncServiceProtocol {
+    func setup() {
+        performSetup()
+    }
+
+    func throttle() {
+        mutex.lock()
+        defer { mutex.unlock() }
+
+        pendingOperationsChainSyncServices.forEach { $0.value.stopSyncUp() }
+        callDataSyncService.stopSyncUp()
+        selectedMetaAccountProvider = nil
+    }
+}
+
+// MARK: - WalletListLocalStorageSubscriber
+
+extension MultisigPendingOperationsSyncService: WalletListLocalStorageSubscriber, WalletListLocalSubscriptionHandler {
+    func handleSelectedWallet(result: Result<ManagedMetaAccountModel?, any Error>) {
+        switch result {
+        case let .success(selectedMetaAccount):
+            mutex.lock()
+            self.selectedMetaAccount = selectedMetaAccount?.info
+            mutex.unlock()
+        case let .failure(error):
+            logger?.error("Failed to fetch selected wallet: \(error)")
+        }
+    }
+}
+
+// MARK: - MultisigCallDataObserver
+
+extension MultisigPendingOperationsSyncService: MultisigCallDataObserver {
+    func didReceive(newCallData: [Multisig.PendingOperation.Key: MultisigCallOrHash]) {
+        mutex.lock()
+        defer { mutex.unlock() }
+
+        knownCallData.merge(
+            newCallData.reduce(into: [:]) { $0[$1.key] = $1.value.call },
+            uniquingKeysWith: { $1 }
+        )
+
+        pendingOperationsChainSyncServices.forEach {
+            $0.value.updatePendingOperations(using: newCallData)
```

Comment:
Why do we need to sync between services passing the newCallData when we have a storage? The service that subscribes by hash can just observe changes and sync only those that missing call data. Am i correct or missing anything?